packages/fuels-test-helpers/src/script.rs

```diff
@@ -1,48 +1,53 @@
 #[cfg(feature = "fuel-core-lib")]
 use fuel_core::service::{Config, FuelService};
 
-#[cfg(feature = "fuel-core-lib")]
-use fuels_signers::provider::Provider;
-
 #[cfg(not(feature = "fuel-core-lib"))]
-use crate::launch_provider_and_get_wallet;
+use crate::node::{Config, FuelService};
+
+use fuels_signers::provider::Provider;
 
 use fuel_gql_client::fuel_tx::{Receipt, Transaction};
 use fuels_contract::script::Script;
+use fuels_core::parameters::TxParameters;
 use fuels_types::errors::Error;
 
 /// Run the Sway script binary located at `binary_filepath` and return its resulting receipts,
 /// without having to setup a node or contract bindings.
 #[allow(dead_code)]
 #[cfg(feature = "fuel-core-lib")]
-pub async fn run_compiled_script(binary_filepath: &str) -> Result<Vec<Receipt>, Error> {
+pub async fn run_compiled_script(
+    binary_filepath: &str,
+    tx_params: TxParameters,
+) -> Result<Vec<Receipt>, Error> {
     let script_binary = std::fs::read(binary_filepath)?;
     let server = FuelService::new_node(Config::local_node()).await.unwrap();
     let provider = Provider::connect(server.bound_address.to_string()).await?;
 
-    let script = get_script(script_binary);
+    let script = get_script(script_binary, tx_params);
 
     script.call(&provider).await
 }
 
 #[allow(dead_code)]
 #[cfg(not(feature = "fuel-core-lib"))]
-pub async fn run_compiled_script(binary_filepath: &str) -> Result<Vec<Receipt>, Error> {
+pub async fn run_compiled_script(
+    binary_filepath: &str,
+    tx_params: TxParameters,
+) -> Result<Vec<Receipt>, Error> {
     let script_binary = std::fs::read(binary_filepath)?;
+    let server = FuelService::new_node(Config::local_node()).await.unwrap();
+    let provider = Provider::connect(server.bound_address.to_string()).await?;
 
-    let wallet = launch_provider_and_get_wallet().await;
-    let provider = wallet.get_provider()?;
-
-    let script = get_script(script_binary);
+    let script = get_script(script_binary, tx_params);
 
-    script.call(provider).await
+    script.call(&provider).await
 }
 
-fn get_script(script_binary: Vec<u8>) -> Script {
+fn get_script(script_binary: Vec<u8>, tx_params: TxParameters) -> Script {
```

Comment:
I think naming this `build_script` would be better. Indeed, what you're doing is bulding a `Script` object from both the script binary and some tx parameters, that we will on a per-tx basis.

Reply:
Yup, ill change that!